pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -242,76 +256,59 @@ func (c *Controller) processPacketCaptureItem() bool {
 
 func (c *Controller) syncPacketCapture(pcName string) error {
-	cleanupStatus := func() {
-		c.mutex.Lock()
-		defer c.mutex.Unlock()
-		state := c.captures[pcName]
-		if state != nil {
-			if state.cancel != nil {
-				state.cancel()
-			}
-			delete(c.captures, pcName)
-		}
-	}
-
 	pc, err := c.packetCaptureLister.Get(pcName)
-	if apierrors.IsNotFound(err) {
+	// List.Get only returns error when the resource is not found.
+	if err != nil {
 		c.cleanupPacketCapture(pcName)
-		cleanupStatus()
 		return nil
 	}
 
 	// Capture will not occur on this Node if a corresponding Pod interface is not found.
 	device := c.getTargetCaptureDevice(pc)
 	if device == "" {
-		klog.V(4).InfoS("Skipping process PacketCapture", "name", pcName)
-		return nil
-	}
-
-	if err := c.validatePacketCapture(&pc.Spec); err != nil {
-		klog.ErrorS(err, "Invalid PacketCapture", "name", pc.Name)
-		if updateErr := c.updateStatus(context.Background(), pcName, packetCaptureState{err: err}); updateErr != nil {
-			klog.ErrorS(err, "Failed to update PacketCapture status", "name", pc.Name)
-		}
-		cleanupStatus()
+		klog.V(4).InfoS("Skipped unrelated PacketCapture", "name", pcName)
 		return nil
 	}
 
-	state := func() packetCaptureState {
+	state, err := func() (packetCaptureState, error) {
 		c.mutex.Lock()
 		defer c.mutex.Unlock()
 		state := c.captures[pcName]
 		if state == nil {
-			state = &packetCaptureState{targetCapturedPacketsNum: pc.Spec.CaptureConfig.FirstN.Number}
+			state = &packetCaptureState{
+				phase:                    packetCapturePhasePending,
+				targetCapturedPacketsNum: pc.Spec.CaptureConfig.FirstN.Number,
+			}
 			c.captures[pcName] = state
 		}
-		phase := state.phase
-		klog.InfoS("Syncing PacketCapture", "name", pcName, "phase", phase)
-		if phase != packetCapturePhasePending {
-			return *state
-		}
 
+		klog.V(2).InfoS("Processing PacketCapture", "name", pcName, "phase", state.phase)
+		if state.phase != packetCapturePhasePending {
+			return *state, nil
+		}
+		// Do not return the error as it's not a transient error.
+		if err := c.validatePacketCapture(&pc.Spec); err != nil {
+			state.captureErr = err
+			return *state, nil
+		}
+		// Return the error as it's a transient error.
 		if c.numRunningCaptures >= maxConcurrentCaptures {
-			err = fmt.Errorf("PacketCapture running count reach limit")
-		} else {
-			// crd spec make sure it's not nil
-			timeout := time.Duration(*pc.Spec.Timeout) * time.Second
-			ctx, cancel := context.WithTimeout(context.Background(), timeout)
-			state.cancel = cancel
-			if err = c.startPacketCapture(ctx, pc, state, device); err != nil {
-				phase = packetCapturePhaseComplete
-			} else {
-				phase = packetCapturePhaseStarted
-				c.numRunningCaptures += 1
-			}
+			state.captureErr = fmt.Errorf("PacketCapture running count reach limit")
+			return *state, state.captureErr
 		}
-		state.phase = phase
-		state.err = err
-		c.captures[pcName] = state
-		return *state
+
+		// crd spec makes sure it's not nil
```

Comment:
The comment is right actually. He means that because of the OpenAPI schema for the CRD, `Spec.Timeout` is guaranteed not to be nil. But maybe the comment could be more accurate.